MetricsUploader/MetricsUploaderWindows.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "MetricsUploader/MetricsUploader.h"
+#include "OrbitBase/Logging.h"
+
+namespace orbit_metrics_uploader {
+
+using orbit_client_protos::OrbitLogEvent;
+using orbit_client_protos::OrbitLogEvent_LogEventType;
+
+constexpr const wchar_t* kMetricsUploaderClientDllName = L"metrics_uploader_client.dll";
+constexpr const char* kSendLogEventFunctionName = "SendOrbitLogEvent";
+constexpr const char* kStartUploaderClientFunctionName = "StartUploaderClient";
+
+MetricsUploader::MetricsUploader() {
+  CHECK(!is_init_);
```

Comment:
I think this can be made more explicit for the user of this class.

You can make the constructor private and add a factory function which returns a `ErrorMessageOr<MetricsUploader>`. So everybody has to go through the factory function and check if they actually got a MetricsUploader object.

A singleton might be another solution.

Reply:
Added `static CreateMetricsUploader()` function and made constructor private.